hash.c

```diff
@@ -1938,28 +1884,14 @@ rb_check_hash_type(VALUE hash)
 
 /*
  *  call-seq:
- *    Hash.try_convert(obj) -> new_hash or nil
- *
- *  Returns the Hash object created by calling <tt>obj.to_hash</tt>:
- *    require 'csv' # => true
- *    row = CSV::Row.new(['Name', 'Age'], ['Bob', 45])
- *    row.respond_to?(:to_hash)  # => true
- *    Hash.try_convert(row) # => {"Name"=>"Bob", "Age"=>45}
- *
- *  Returns the given <tt>obj</tt> if it is a Hash.
+ *    Hash.try_convert(obj) -> obj, new_hash, or nil
  *
- *  Returns <tt>nil</tt> unless <tt>obj.respond_to?(:to_hash)</tt>.
+ *  If +obj+ is a \Hash object, returns +obj+.
  *
- *  ---
+ *  Otherwise if +obj+ responds to <tt>:to_hash</tt>,
+ *  calls <tt>obj.to_hash</tt> and returns the result.
  *
- *  Raises an exception unless <tt>obj.to_hash</tt> returns a \Hash object:
```

Comment:
I would keep this line but move it to the bottom (still drop the example).

Reply:
Fixed.